base/interactiveutil.jl

```diff
@@ -254,8 +254,13 @@ function gen_call_with_extracted_types(fcn, ex0)
                     Expr(:call, typesof, map(esc, args[2:end])...))
     end
     exret = Expr(:none)
+    is_macro = false
     ex = expand(ex0)
-    if !isa(ex, Expr)
+    if fcn in [:which, :less, :edit, :functionloc] && ex0.head == :macrocall
```

Comment:
Trust me, it's not going to be "uglier" than the AST of this function (`gen_call_with_extracted_types`) itself.